tools/cli/helpers/analytics.js

```diff
@@ -0,0 +1,80 @@
+import crypto from 'crypto';
+import Configstore from 'configstore';
+import enquirer from 'enquirer';
+
+const configStore = new Configstore( 'automattic/jetpack-cli/cli' );
+const baseEventName = 'jetpack_cli_'; // Please use 'jetpack_test_cli_' when testing new Tracks events.
+
+/**
+ * Checks if:
+ * - the user has been asked to enable analytics tracking.
+ * - the user has enabled or disabled analytics tracking.
+ * - the user has a UUID set.
+ *
+ * @returns {boolean} Whether analytics tracking is enabled.
+ */
+export async function checkAnalyticsEnabled() {
+	if ( process.env.CI ) {
+		return false;
+	}
+
+	if ( configStore.get( 'askedToEnableAnalytics' ) === undefined ) {
```

Comment:
Re the silly behavior mentioned above, if we want to care we might do like this:
```suggestion
	if ( configStore.get( 'askedToEnableAnalytics' ) === undefined ) {
		// Skip asking if they're trying to run `jetpack cli analytics` itself.
		const args = process.argv.filter( arg => ! arg.startsWith( '-' ) );
		if ( args[2] === 'cli' && args[3] === 'analytics' ) {
			return false;
		}

```

Although, to follow up on that, we should then move lines 31–37 and 40–42 into a separate function to also call from `tools/cli/commands/cli.js`, instead of partially duplicating the code there. We might want to do that anyway.

Reply:
Thanks, that would be a bit cleaner, done in bd0aaa7